File: spring-boot-webmvc-tests/src/main/java/de/rieckpil/learning/SecurityConfig.java
package de.rieckpil.learning;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;

@Configuration
public class SecurityConfig extends WebSecurityConfigurerAdapter {

  @Override
  protected void configure(HttpSecurity http) throws Exception {
    http.authorizeRequests(authorize -> authorize
      .mvcMatchers("/sample", "/sample/xml").permitAll()
      .mvcMatchers("/sample/secure").hasRole("ADMIN")
      .anyRequest().authenticated())
      .cors().disable();
  }
}

File: spring-boot-kotlin-dsl/src/main/kotlin/de/rieckpil/learning/SecurityConfig.kt
package de.rieckpil.learning

import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.dsl.config.builders.servlet.invoke

@Configuration
class SecurityConfig : WebSecurityConfigurerAdapter() {
  override fun configure(http: HttpSecurity?) {
    http {
      authorizeRequests {
        authorize(anyRequest, permitAll)
      }
    }
  }
}
